code/game/objects/structures/fireaxe.dm

```diff
@@ -6,20 +6,38 @@
 	icon_state = "fireaxe"
 	anchored = 1
 	density = 0
-	armor = list(melee = 50, bullet = 20, laser = 0, energy = 100, bomb = 10, bio = 100, rad = 100, fire = 0, acid = 0)
+	armor = list(melee = 50, bullet = 20, laser = 0, energy = 100, bomb = 10, bio = 100, rad = 100, fire = 90, acid = 50)
 	var/locked = 1
 	var/open = 0
-	var/health = 60
+	obj_integrity = 150
+	max_integrity = 150
+	integrity_failure = 50
 
 /obj/structure/fireaxecabinet/New()
 	..()
 	update_icon()
 
+/obj/structure/fireaxecabinet/Destroy()
+	if(fireaxe)
+		qdel(fireaxe)
+		fireaxe = null
+	return ..()
+
 /obj/structure/fireaxecabinet/attackby(obj/item/I, mob/user, params)
 	if(iscyborg(user) || istype(I,/obj/item/device/multitool))
 		toggle_lock(user)
-		return
-	if(open || health <= 0)
+	else if(istype(I, /obj/item/stack/sheet/glass) && broken)
+		var/obj/item/stack/sheet/glass/G = I
+		if(G.get_amount() < 2)
+			user << "<span class='warning'>You need two glass sheets to fix [src]!</span>"
+			return
+		user << "<span class='notice'>You start fixing [src]...</span>"
+		if(do_after(user, 20, target = src))
```

Comment:
`if(do_after(user, 20, target = src) && G.use(2))` so that it fails if you use the glass on something else and take it down to one sheet